src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,81 @@
+package duke;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.LocalDateTime;
+
+/**
+ * Encapsulates a class which interacts with the output file to save data after
+ */
+public class Storage {
+    private File txt;
+
+    /**
+     * Loads file to save task list data
+     *
+     * @throws Exception IO or Security Exception thrown by createNewFile
+     */
+    public void loadFile() throws Exception {
+        Path p = Paths.get("data");
+        if (!Files.exists(p)) {
+            Files.createDirectories(p);
+        }
+        File f = Paths.get("data", "duke.txt").toFile();
+        f.createNewFile();
+        txt = f;
+    }
+
+    /**
+     * Reads the output file and loads any saved tasks
+     *
+     * @param t the TaskList object to save the list of tasks
+     */
+    public void readFromFile(TaskList t) {
+        try {
+            BufferedReader br = new BufferedReader(new FileReader(txt));
+            String curLine;
+            while ((curLine = br.readLine()) != null) {
+                String[] parts = curLine.split(" ", 3);
+                String[] descriptionParts;
+                char taskType = parts[0].charAt(1);
+                boolean completionStatus = parts[1].charAt(1) == 'X';
```

Comment:
`isCompleted` might make for a more understandable boolean variable!! :)

Reply:
Fixed, Thanks for your help